### DISCUSSION TITLE
harbor command line interface

### DISCUSSION THREAD

**Discussion by shahidv3** [goharbor/harbor#14438] [Ideas]
Hi Team

I would like to develop and contribute  harbor command line interface utility .
please share your thoughts.

thanks ,sahid

**Comment by steven-zou** [comment]
Share some background:

Before Harbor V1.8, one contributor has tried to develop a CLI (golang based) but not continued.
Now, a contributor from VMware did something based on Harbor's swagger API with java lang (it will run as a separate project in another namespace, not under `goharbor` namespace).

**Comment by steven-zou** [comment]
I think a well-defined CLI is welcome.

**Reply by shahidv3** [reply]
Thanks @steven-zou for the reply

The command line interface is developed using Python and its developed against version 2.2.0


This will be useful for any DevOps or Developer to automate the admin operations or any housekeeping activities or Day2 operations of Harbor 

thanks , sahid
